src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs

```diff
@@ -84,17 +84,30 @@ public async Task<PackageAnalysisState> AnalyzeAsync(IProject project, PackageAn
         private async Task<NuGetReference?> GetUpdatedPackageVersionAsync(NuGetReference packageReference, IEnumerable<TargetFrameworkMoniker> targetFramework, CancellationToken token)
         {
             var latestMinorVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, true, token).ConfigureAwait(false);
+            NuGetReference? prereleaseCandidate = null;
 
             foreach (var newerPackage in latestMinorVersions)
```

Comment:
Does this say we upgrade packages to the lowest version that supports the target TFM?

I thought we upgraded to the latest package always.

Reply:
We upgrade to the latest minor/build version of the lowest major version that supports the target TFM. We do that to reduce breaking changes (by using the lowest major version) but maximize bug/security fixes (by using the highest minor version).

Also, after this PR, we don't use pre-release versions unless they're the only ones that support the target TFM.